Index: python/gui/qgslegendinterface.sip
===================================================================
--- python/gui/qgslegendinterface.sip	(revision 15533)
+++ python/gui/qgslegendinterface.sip	(working copy)
@@ -50,7 +50,7 @@
   public slots:
 
     //! Add a new group
-    virtual int addGroup( QString name, bool expand = true ) =0;
+    virtual int addGroup( QString name, bool expand = true, bool forceAtEnd = false ) =0;
 
     //! Remove group on index
     virtual void removeGroup( int groupIndex ) =0;
Index: src/app/legend/qgsapplegendinterface.h
===================================================================
--- src/app/legend/qgsapplegendinterface.h	(revision 15533)
+++ src/app/legend/qgsapplegendinterface.h	(working copy)
@@ -65,7 +65,7 @@
   public slots:
 
     //! Add a new group
-    int addGroup( QString name, bool expand = true );
+    int addGroup( QString name, bool expand = true, bool forceAtEnd = false );
 
     //! Remove all groups with the given name
     void removeGroup( int groupIndex );
Index: src/app/legend/qgslegend.cpp
===================================================================
--- src/app/legend/qgslegend.cpp	(revision 15533)
+++ src/app/legend/qgslegend.cpp	(working copy)
@@ -151,11 +151,15 @@
   emit currentLayerChanged( layer );
 }
 
-int QgsLegend::addGroup( QString name, bool expand )
+int QgsLegend::addGroup( QString name, bool expand, bool forceAtEnd )
 {
   if ( name.isEmpty() )
     name = tr( "group" ); // some default name if none specified
 
+  if ( forceAtEnd )
+	//forces the group to be created at the end of the list 
+	//and not in the currently selected group
+	setCurrentItem( 0 );
   QgsLegendGroup *parent = dynamic_cast<QgsLegendGroup *>( currentItem() );
 
   QgsLegendGroup *group;
Index: src/app/legend/qgslegend.h
===================================================================
--- src/app/legend/qgslegend.h	(revision 15533)
+++ src/app/legend/qgslegend.h	(working copy)
@@ -221,7 +221,7 @@
      * @param expand expand the group
      * @return void
      */
-    int addGroup( QString name = QString(), bool expand = true );
+    int addGroup( QString name = QString(), bool expand = true, bool forceAtEnd = false );
 
     /*!
      * Removes all groups with the given name.
Index: src/app/legend/qgsapplegendinterface.cpp
===================================================================
--- src/app/legend/qgsapplegendinterface.cpp	(revision 15533)
+++ src/app/legend/qgsapplegendinterface.cpp	(working copy)
@@ -31,9 +31,9 @@
 {
 }
 
-int QgsAppLegendInterface::addGroup( QString name, bool expand )
+int QgsAppLegendInterface::addGroup( QString name, bool expand, bool forceAtEnd )
 {
-  return mLegend->addGroup( name, expand );
+  return mLegend->addGroup( name, expand, forceAtEnd );
 }
 
 void QgsAppLegendInterface::removeGroup( int groupIndex )
Index: src/gui/qgslegendinterface.h
===================================================================
--- src/gui/qgslegendinterface.h	(revision 15533)
+++ src/gui/qgslegendinterface.h	(working copy)
@@ -80,7 +80,8 @@
   public slots:
 
     //! Add a new group
-    virtual int addGroup( QString name, bool expand = true ) = 0;
+    //! forceAtEnd forces the new group to be created at the end of the legend
+    virtual int addGroup( QString name, bool expand = true, bool forceAtEnd = false ) = 0;
 
     //! Remove group on index
     virtual void removeGroup( int groupIndex ) = 0;
